src/win/fs.c

```diff
@@ -1701,21 +1702,43 @@ static void fs__symlink(uv_fs_t* req) {
   WCHAR* new_pathw = req->fs.info.new_pathw;
   int flags = req->fs.info.file_flags;
   int result;
+  int err;
 
 
   if (flags & UV_FS_SYMLINK_JUNCTION) {
     fs__create_junction(req, pathw, new_pathw);
-  } else if (pCreateSymbolicLinkW) {
-    result = pCreateSymbolicLinkW(new_pathw,
-                                  pathw,
-                                  flags & UV_FS_SYMLINK_DIR ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0) ? 0 : -1;
-    if (result == -1) {
-      SET_REQ_WIN32_ERROR(req, GetLastError());
-    } else {
-      SET_REQ_RESULT(req, result);
-    }
-  } else {
+    return;
+  }
+  if (!pCreateSymbolicLinkW) {
     SET_REQ_UV_ERROR(req, UV_ENOSYS, ERROR_NOT_SUPPORTED);
+    return;
+  }
+
+  if (flags & UV_FS_SYMLINK_DIR) {
+    flags = SYMBOLIC_LINK_FLAG_DIRECTORY;
+  } else {
+    flags = uv__file_symlink_usermode_flag;
+  }
+
+  result = pCreateSymbolicLinkW(new_pathw, pathw, flags);
+  if (result != 0) {
```

Comment:
Very minor, but if you inline the `pCreateSymbolicLinkW()` call here, you can get rid of `result` completely.